src/components/EvolutionTab.js

```diff
@@ -0,0 +1,117 @@
+import React, { useState, useEffect } from "react";
+import { fetchPokemon } from "../util/fetchPokemonData";
+
+const EvolutionTab = ({ evolutionChainUrl }) => {
+  const [evolutionNamesObjects, setEvolutionNamesObjects] = useState([]);
+  const [evolutionNameStrings, setEvolutionNameStrings] = useState([]);
+  const [spriteUrls, setSpriteUrls] = useState([]);
+
+  let evolutionNamesList = [];
+  let chain = {};
+
+  // Function to fetch chain object, save to state hook
+  const fetchChain = async () => {
+    try {
+      const chainData = await fetch(evolutionChainUrl)
+        .then((response) => response.json())
+        .then((data) => {
+          chain = data.chain;
+        });
+      evolutionNamesList = getEvolutionNames(chain);
```

Comment:
I think the `chainData` receives the data after the final `then` method? On line 17, you could `return data.chain` which is stored in `chainData`, and then on line 20 you could do `getEvolutionNames(chainData)`. But, if it works then it works 😘!

Reply:
On line 20, getEvolutionNames(chain) returns an array that we save to a variable for another function. Not sure if there's a convention for this, but I thought it would be weird to put that declaration inside the last .then().